ld fails because of multiple definition of `camlFlags' in opam-file-format and core

Hello, I think I found something wrong with core or opam-format.

I'm developing a software, [Satyrographos](https://github.com/na4zagin3/satyrographos), that depends on both `core` and `opam-format`.

The build started to fail with a link error due to duplicated symbols camlFlags and so on a few days ago. (See https://travis-ci.com/github/na4zagin3/satyrographos/jobs/443392985#L3291)

Here is a minimal reproducible example.

```ocaml
(* main.ml *)
open Core

let () =
  OpamFile.OPAM.empty
  |> OpamFile.OPAM.write_to_string
  |> print_endline
```

```dune
;;; dune
(executable
 (public_name mre)
 (name main)
 (libraries core opam-format))
```

```dune
;;; dune-project
(lang dune 2.7)
(name mre)
```

```sh
# install dependencies with opam-file-format 2.1.0
$ opam switch create test 4.11.1
$ opam install --yes dune.2.7.1 core.v0.14.0 core_kernel.v0.14.1 opam-format.2.0.7 opam-file-format.2.1.1
# the example does not compile
$ dune build
    ocamlopt main.exe (exit 2)

...

/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__data_begin':
:(.data+0x0): multiple definition of `camlFlags__data_begin'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.data+0x0): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__create_4091':
/home/na4zagin3/.opam/4.11.1/.opam-switch/build/core_kernel.v0.14.1/_build/default/flags/src/flags.ml:5: multiple definition of `camlFlags__code_begin'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.text+0x0): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags':
:(.data+0x2c8): multiple definition of `camlFlags'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.data+0x8): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__gc_roots':
:(.data+0x2d8): multiple definition of `camlFlags__gc_roots'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.data+0x8): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__entry':
/home/na4zagin3/.opam/4.11.1/.opam-switch/build/core_kernel.v0.14.1/_build/default/flags/src/flags.ml:127: multiple definition of `camlFlags__entry'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.text+0x0): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__code_end':
:(.text+0x17db): multiple definition of `camlFlags__code_end'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.text+0x5c): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__data_end':
:(.data+0x6d0): multiple definition of `camlFlags__data_end'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.data+0x58): first defined here
/usr/bin/ld: /home/na4zagin3/.opam/4.11.1/lib/core_kernel/flags/flags.a(flags.o): in function `camlFlags__frametable':
:(.data+0x6d8): multiple definition of `camlFlags__frametable'; /home/na4zagin3/.opam/4.11.1/lib/opam-file-format/opam_file_format.a(flags.o)::(.data+0x60): first defined here
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking (exit code 1)
```

## Environment

Ubuntu 20.04 (x86_64)

```
$ opam --version
2.0.5
$ opam list -i '*'
# Packages matching: installed & name-match(*)
# Name                  # Installed # Synopsis
base                    v0.14.0     Full standard library replacement for OCaml
base-bigarray           base
base-bytes              base        Bytes library distributed with the OCaml compiler
base-threads            base
base-unix               base
base_bigstring          v0.14.0     String type based on [Bigarray], for use in I/O and C-bindings
base_quickcheck         v0.14.0     Randomized testing framework, designed for compatibility with 
bin_prot                v0.14.0     A binary protocol generator
conf-m4                 1           Virtual package relying on m4
core                    v0.14.0     Industrial strength alternative to OCaml's standard library
core_kernel             v0.14.1     Industrial strength alternative to OCaml's standard library
cppo                    1.6.6       Code preprocessor like cpp for OCaml
csexp                   1.3.2       Parsing and printing of S-expressions in Canonical form
dune                    2.7.1       Fast, portable, and opinionated build system
dune-configurator       2.7.1       Helper library for gathering system configuration
fieldslib               v0.14.0     Syntax extension to define first class values representing rec
jane-street-headers     v0.14.0     Jane Street C header files
jst-config              v0.14.0     Compile-time configuration for Jane Street libraries
num                     1.4         The legacy Num library for arbitrary-precision integer and rat
ocaml                   4.11.1      The OCaml compiler (virtual package)
ocaml-base-compiler     4.11.1      Official release 4.11.1
ocaml-compiler-libs     v0.12.3     OCaml compiler libraries repackaged
ocaml-config            1           OCaml Switch Configuration
ocaml-migrate-parsetree 2.1.0       Convert OCaml parsetrees between different versions
ocamlfind               1.8.1       A library manager for OCaml
ocamlgraph              2.0.0       A generic graph library for OCaml
octavius                1.2.2       Ocamldoc comment syntax parser
opam-core               2.0.7
opam-file-format        2.0.0       Parser and printer for the opam file syntax
opam-format             2.0.7
parsexp                 v0.14.0     S-expression parsing library
ppx_assert              v0.14.0     Assert-like extension nodes that raise useful errors on failur
ppx_base                v0.14.0     Base set of ppx rewriters
ppx_bench               v0.14.1     Syntax extension for writing in-line benchmarks in ocaml code
ppx_bin_prot            v0.14.0     Generation of bin_prot readers and writers from types
ppx_cold                v0.14.0     Expands [@cold] into [@inline never][@specialise never][@local
ppx_compare             v0.14.0     Generation of comparison functions from types
ppx_custom_printf       v0.14.0     Printf-style format-strings for user-defined string conversion
ppx_derivers            1.2.1       Shared [@@deriving] plugin registry
ppx_enumerate           v0.14.0     Generate a list containing all values of a finite type
ppx_expect              v0.14.0     Cram like framework for OCaml
ppx_fields_conv         v0.14.1     Generation of accessor and iteration functions for ocaml recor
ppx_fixed_literal       v0.14.0     Simpler notation for fixed point literals
ppx_hash                v0.14.0     A ppx rewriter that generates hash functions from type express
ppx_here                v0.14.0     Expands [%here] into its location
ppx_inline_test         v0.14.1     Syntax extension for writing in-line tests in ocaml code
ppx_jane                v0.14.0     Standard Jane Street ppx rewriters
ppx_js_style            v0.14.0     Code style checker for Jane Street Packages
ppx_let                 v0.14.0     Monadic let-bindings
ppx_module_timer        v0.14.0     Ppx rewriter that records top-level module startup times
ppx_optcomp             v0.14.0     Optional compilation for OCaml
ppx_optional            v0.14.0     Pattern matching on flat options
ppx_pipebang            v0.14.0     A ppx rewriter that inlines reverse application operators `|>`
ppx_sexp_conv           v0.14.1     [@@deriving] plugin to generate S-expression conversion functi
ppx_sexp_message        v0.14.0     A ppx rewriter for easy construction of s-expressions
ppx_sexp_value          v0.14.0     A ppx rewriter that simplifies building s-expressions from oca
ppx_stable              v0.14.1     Stable types conversions generator
ppx_string              v0.14.1     Ppx extension for string interpolation
ppx_typerep_conv        v0.14.1     Generation of runtime types from type declarations
ppx_variants_conv       v0.14.1     Generation of accessor and iteration functions for ocaml varia
ppxlib                  0.17.0      Standard library for ppx rewriters
re                      1.9.0       RE is a regular expression library for OCaml
result                  1.5         Compatibility Result module
seq                     base        Compatibility package for OCaml's standard iterator type start
sexplib                 v0.14.0     Library for serializing OCaml values to and from S-expressions
sexplib0                v0.14.0     Library containing the definition of S-expressions and some ba
spawn                   v0.13.0     Spawning sub-processes
splittable_random       v0.14.0     PRNG that can be split into independent streams
stdio                   v0.14.0     Standard IO library for OCaml
stdlib-shims            0.1.0       Backport some of the new stdlib features to older compiler
time_now                v0.14.0     Reports the current time
timezone                v0.14.0     Time-zone handling
typerep                 v0.14.0     Typerep is a library for runtime types
variantslib             v0.14.0     Part of Jane Street's Core library
```

--- a/src/dune
+++ b/src/dune
@@ -3,6 +3,7 @@
   (public_name opam-file-format)
   (synopsis "Parser and printer for the opam file syntax")
   (wrapped false)
+  (modules :standard \ flags)
   (flags :standard (:include flags.sexp))
   (modules_without_implementation opamParserTypes))
 

